### ISSUE TITLE
Vimeo embed not working

### ISSUE THREAD

**Issue by DominicBoettger** [Simbul/baker#510]
Hi,

i have a strange problem. With iOS5 i had no issues when embedding vimeo iframe links. Since upgrading to the newest iOS6 release it's not possible to embed vimeo videos.
The version compiled with the iOS5 SDK is still working, but if i compile a new release the videos are not working anymore

The video's preview picture is displayed, but the video is not playing when pressing the play button. The loading bar is animated but the video does not start playing.

Do you have any idea what causes this problem and how I can solve it?

Thanks!

Regards

Dominic

**Comment by folletto** [comment]
Did you test it on Mobile Safari as well? Does it work there?

**Comment by DominicBoettger** [comment]
Tried it on mobile safari without any issues, but when i tried to view the video in chrome for iOS i have the same issue after trying to play the video a second time.

**Comment by DominicBoettger** [comment]
Tried it again with chrome and different baker versions. Older versions of baker works fine, only the latest versions is buggy. Chrome does not have the issue anymore.

Any ideas? There are too many changes in baker code and I have no idea where to start debugging.

**Comment by folletto** [comment]
Given your feedback and the fact the problem doesn't appear consistently I think it's outside Baker's scope unfortunately, we don't have any code that deals with that, it's the `UIWebView`... but...

However: did you actually compile older versions of Baker against iOS6 and it worked, did I get this correctly? If you did that, at which version did it stop working?

**Comment by DominicBoettger** [comment]
I did some tests again and chrome doesn't have the problem (anymore?). With baker version 3.2 it works, but with 3.2.2 videos are not played.

**Comment by folletto** [comment]
Sorry if I look pedantic but you are saying that Baker 3.2 compiled fresh on iOS6 still works, did I get that right? :)

**Comment by DominicBoettger** [comment]
Yes, that is correct. If you create an application under 3.2.2 with the template book and just add the vimeo embed code. The video is not playing.

**Comment by folletto** [comment]
Ok, thank you. We'll look into it. :)

**Comment by DominicBoettger** [comment]
Youtube plays, but pausing the video is not possible. The controls doesn't seem to work.
So there is currently no solution i know to embed video - URL's under baker.

Any ideas?

**Comment by folletto** [comment]
Well the "solution" would be to use the iOS standard... so the standard video tag and simply embed the videos. But of course we'll have to look into this. Or if someone wants to look into this and provide a patch, of course. ;)

**Comment by DominicBoettger** [comment]
You mean the normal <video> tag?

**Comment by folletto** [comment]
Lol yes, sorry github comment ate the tag :P

**Comment by f3r3nc** [comment]
@DominicBoettger could you share the hpub you are using for this? Or create an example that is not working?
Although it was not Baker, the last time I've included a video in an embedded UIWebView I used youtube and the **flash** link as there were problems with the HTML5 video tags.

**Comment by DominicBoettger** [comment]
Ah I found that the video tag is a much better solution for me. Added the vimeo files to the video tag. Tried also some other sources. In safari the video tag play the audio too, but not in baker. Any ideas?

**Comment by DominicBoettger** [comment]
Audio problem solved.

https://github.com/Simbul/baker/pull/521

**Comment by Simbul** [comment]
Pull request merged. Thanks for the contribution!

Can we consider this issue closed, then?

**Comment by Simbul** [comment]
Actually, I've just run a test by adding a Vimeo `iframe` to a page and it does indeed not play.

A solution may be the one detailed in [this comment](http://stackoverflow.com/questions/12648014/ios-6-not-playing-vimeo-embedded-videos#comment17318693_12648014) on StackOverflow. Haven't tested it, but it may be something worth trying.

**Comment by DominicBoettger** [comment]
Hi Simbul,

this pull request does not solve the iframe problems, but solves the problem that videos were not able to be played at all under iOS6.

If you create a video tag and add a URL ( for example to the m4v on your vimeo account ) you were not able to hear any sound on video playback.

The pull requests solves the issue that under iOS6 you need to create a audio session before instantiating a UIWebView.

I tried everything to get a running iFrame player under baker with iOS6, but this does not seem to be possible.

Regards

Dominic

**Comment by Simbul** [comment]
So not even the StackOverflow comment I posted above solved the iframe issue for you?

**Comment by DominicBoettger** [comment]
No, it did not work. iFrame implementations are buggy und iOS6. Solved it using the vimeo video URLS and the html5 video tag. Works much more stable than any iFrame implementation.

Thanks for your help!

Regards
Dominic

**Comment by roninboy** [comment]
I found that if I remove the User Agent references from the BakerAppDelegate.m (which are determining the new Youtube player options) - you can revert to the old flash embed code (which although deprecated) will still work.
The old player will allow fullscreen in any orientation. It also avoids all the issues with audio sessions, pausing problems etc.

**Comment by brianharrington** [comment]
@DominicBoettger, could you post the code your using. I am having the same troubles.

**Comment by DominicBoettger** [comment]
@brianharrington Of course! :-)

``` html
<div class="video-container">
    <video controls="true" poster="https://secure-b.vimeocdn.com/ts/352/839/352839243_1280.jpg">
        <source src="http://player.vimeo.com/external/51197485.hd.mp4?s=2eda6c7189a461f9941cc74ef584c2b5" type="video/mp4"/>
    </video>
</div>
```

**Comment by Simbul** [comment]
Ok, so it looks like we have a solution here: using `<video>` tags instead of an iframe.

@brianharrington let us know if this fixes your issue :)

**Comment by yitweng** [comment]
DominicBoettger , sorry to ask maybe the obvious, how do you get the URL of the video off Vimeo. I search and hunt for it in the page sources and couldn't find any lead.

Thanks.

Jon

**Comment by folletto** [comment]
Did you try removing the User Agent like they say in topic #611 from `AppDelegate.m`? :)

**Comment by DominicBoettger** [comment]
@yitweng Not sure if a pro account is needed but I am able to get the urls under the video settings.

**Comment by yitweng** [comment]
@DominicBoettger i think you are right about the pro account, cos mine isn't. @folletto removing user agent dont solve the iframe issue for vimeo :( still not able to play back the video :(

**Comment by DominicBoettger** [comment]
Will this help? http://stackoverflow.com/questions/4460394/how-can-i-find-download-links-for-vimeo-videos

**Comment by folletto** [comment]
A few months has passed... did anyone solve this in the meantime? Can someone feedback about this? :)

**Comment by anjimi** [comment]
Just came across this, and would like to note that we have a vimeo video in our latest publication and it works perfectly. Used the standard iframe embed code from vimeo. Baker version 4.1, I have removed the user agent from AppDelegate.m to make youtube/vimeo videos use the same player as included video files.
